mbuild/compound.py

```diff
@@ -177,7 +179,8 @@ def load(filename_or_object, relative_to_module=None, compound=None, coords_only
         compound.from_pybel(mymol, infer_hierarchy=infer_hierarchy)
 
     else:
-        traj = md.load(filename_or_object, **kwargs)
+        mdtraj = import_('mdtraj')
+        traj = mdtraj.load(filename_or_object, **kwargs)
         compound.from_trajectory(traj, frame=-1, coords_only=coords_only,
                 infer_hierarchy=infer_hierarchy)
```

Comment:
I think this could be a better version of this:
```python3
else:
    if has_mdtraj:
        mdtraj = import_('mdtraj')
        try:
            traj = mdtraj.load(filename_or_object, **kwargs)
            compound.from_trajectory(traj, frame=-1, coords_only=coords_only,
                    infer_hierarchy=infer_hierarchy)
        except:
             raise TypeError("Unrecognized file format")
    else:
        raise TypeError("Unrecognized file format")
```

Otherwise, if you try to load in an unsupported format, it will give an MDTraj error (if mdtraj is installed), or a `please install mdtraj` error message.